### PR TITLE
PLAT-1151 High-Level Input Customization Methods on EmfForeignEntityAttribute

### PR DESCRIPTION
**Pull request by oliver-richers**

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java [hunk 1]
```diff
@@ -40,8 +40,7 @@ public void customizeAttributeProperties(IEmfAttributeList<AGModuleInstancePermi
 		attributes//
 			.editAttribute(AGModuleInstancePermissionAssignment.USER)
 			.setImmutable(true)
-			.setPredicateMandatory(EmfPredicates.always())
-			.setInputFactory(ModuleInstancePermissionAssignmentUserInput::new);
+			.setPredicateMandatory(EmfPredicates.always());
```
Comment (daniel-klose) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java:43]: Shouldnt there be a `.addFilter(user -> !user.isSystemUser());` here?

File: platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/user/configuration/AGWorkflowUserConfigurationTable.java [hunk 2]
```diff
@@ -51,8 +50,8 @@ public void customizeLoggers(EmfChangeLoggerSet<AGWorkflowUserConfiguration> log
 	public void customizeAttributeProperties(IEmfAttributeList<AGWorkflowUserConfiguration> attributes) {
 
 		attributes//
-			.editAttribute(AGWorkflowUserConfiguration.SUBSTITUTE)
-			.setInputFactory(UserInput::new);
```
Comment (daniel-klose) [inline platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/user/configuration/AGWorkflowUserConfigurationTable.java:55]: Can we remove the `UserInput` class?
It seems orphaned now, and a bit useless in general

**Review by daniel-klose — CHANGES_REQUESTED** [review]

File: platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/user/configuration/AGWorkflowUserConfigurationTable.java (hunk 2 above)
Comment (oliver-richers) [reply platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/user/configuration/AGWorkflowUserConfigurationTable.java:55]: unfortunately, we cannot, it is use in EAS aside from EMF-attributes

File: platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java (hunk 1 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java:43]: hmm, not sure if really necessary, but I can do that

Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java:44]: ok, added, also added to `AGRoleUser` for symmetry

**Review by daniel-klose — APPROVED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java:44]: I don't see the point of suppressing the system user only in two distinct places. What about the remaining 5000 user inputs, particularly in the payload software? IMO, we either need a systematic solution to this - or none.

Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/permission/assignment/AGModuleInstancePermissionAssignmentTable.java:44]: not sure if this has anything to do with this PR? I just retained the status quo and extended it to `AGRoleUser` for symmetry...so?

**Pull request merged**